### PR TITLE
Add resolvepath macro

### PR DESCRIPTION
**Pull request by tobibeer**
relates to #1387

* **$:/core/ui/ControlPanel/Saving** — modified to use the below macro
* **$:/core/modules/macros/serverpath.js** — macro constructing the url
* **$:/language/ControlPanel/Saving/TiddlySpot/Hint** — added example

### CODE REVIEW

**Comment by tobibeer** [comment]
demo (temporary): http://goto.bplaced.net/tw5/5.1.7.htm

**Comment by Jermolene** [comment]
Hi @tobibeer interesting. I don't like the specificity of the new serverpath macro though. Can we perhaps make it match the semantics of Node's path.resolve() function?

Also it looks like the backup path is no longer rendered as a link, is that right?

**Comment by tobibeer** [comment]
> I don't like the specificity of the new serverpath macro though. Can we perhaps make it match the semantics of Node's path.resolve() function?

You mean replicate or include that path module? I'd be inclined to leave such things to you as I imagine you have a better overall idea on where and how a path module will be required / used / implemented.

> Also it looks like the backup path is no longer rendered as a link, is that right?

The link to the backup folder is what this pull request is about. I am seeing a working link in both cases, TiddlySpot and custom script. What are you seeing? Before there was no link for a custom **store.php**.

**Comment by Jermolene** [comment]
> The link to the backup folder is what this pull request is about. I am seeing a working link in both cases, TiddlySpot and custom script

My mistake, I misread the code.

> You mean replicate or include that path module

We've got a compatible implementation in `$tw.utils.resolvePath` that works in the browser. If that were exposed as a macro would you be able to use it here?

**Comment by tobibeer** [comment]
> We've got a compatible implementation in $tw.utils.resolvePath that works in the browser. If that were exposed as a macro would you be able to use it here?

Oh, did not see that. It's precisely what's needed. Made a new commit.

File: core/modules/macros/resolvePath.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/*\
+title: $:/core/modules/macros/resolvePath.js
```
Comment (Jermolene) [inline core/modules/macros/resolvePath.js]: Again, for consistency the title and filename should be "resolvepath".

Comment (tobibeer) [inline core/modules/macros/resolvePath.js]: sure

**Comment by tobibeer** [comment]
@Jermolene renamed + added documntation

**Comment by Jermolene** [comment]
Looks good thank you @tobibeer

File: core/language/en-GB/ControlPanel.multids [hunk 2]
```diff
@@ -56,7 +56,7 @@ Saving/TiddlySpot/Backups: Backups
 Saving/TiddlySpot/Description: These settings are only used when saving to http://tiddlyspot.com or a compatible remote server
 Saving/TiddlySpot/Filename: Upload Filename
 Saving/TiddlySpot/Heading: ~TiddlySpot
-Saving/TiddlySpot/Hint: //The server URL defaults to `http://<wikiname>.tiddlyspot.com/store.cgi` and can be changed to use a custom server address//
+Saving/TiddlySpot/Hint: //The server URL defaults to `http://<wikiname>.tiddlyspot.com/store.cgi` and can be changed to use a custom server address, e.g. `http://me.com/store.php`.//
```
Comment (pmario) [inline core/language/en-GB/ControlPanel.multids]: me.com actually is a valid URL and belongs to apple. It's forwarded to icloud. ... So please use example.com which is there for exactly that purpose.

File: editions/tw5.com/tiddlers/macros/ResolvePath.tid [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+! Examples
+
+A trivial example to show how the macro works:
+
+```
+<<resolvepath "./backup" "http://mysite.com/store.php">>
+```
+
+<<resolvepath "./backup" "http://mysite.com/store.php">>
```
Comment (pmario) [inline editions/tw5.com/tiddlers/macros/ResolvePath.tid]: example.com ... mysite.com is an existing URL

File: core/language/en-GB/ControlPanel.multids (hunk 2 above)
Comment (tobibeer) [inline core/language/en-GB/ControlPanel.multids]: Would you mind making a pull request with whatever you recommend?

Comment (pmario) [inline core/language/en-GB/ControlPanel.multids]: sure. I kind of ignored, that it's already merged :)

Comment (tobibeer) [inline core/language/en-GB/ControlPanel.multids]: :+1:  ^^

**Pull request merged**